presto-product-tests/src/main/java/com/facebook/presto/tests/ImpersonationTests.java

```diff
@@ -0,0 +1,96 @@
+
+import static com.facebook.presto.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE_HDFS_IMPERSONATION;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE_KERBEROS_HDFS_IMPERSONATION;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE_KERBEROS_HDFS_NO_IMPERSONATION;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+@Requires(ImmutableNationTable.class)
+public class ImpersonationTests
+        extends ProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+    private static final String tableName = "checkTableOwner";
+    private static final String tablepath = "/user/hive/warehouse/checktableowner";
+
+    @Inject
+    @Named("databases.presto.jdbc_user")
+    private String jdbcuserkerberos;
+
+    @Inject
+    @Named("databases.presto.jdbc_user")
+    private String jdbchdfsimpersonation;
+
+    // Without impersonation, the table is created with user specified for
+    // -DHADOOP_USER_NAME in jvm.config
+    @Test(groups = {SINGLENODE, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNode()
+            throws Exception
+    {
+        checkTableOwner("hive");
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    @Test(groups = {SINGLENODE_HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeHdfsImpersonation()
+            throws Exception
+    {
+        requireNonNull(jdbchdfsimpersonation, "databases.presto.jdbc_user is null");
+        checkTableOwner(jdbchdfsimpersonation);
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    @Test(groups = {SINGLENODE_KERBEROS_HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeKerberosHdfsImpersonation()
+            throws Exception
+    {
+        requireNonNull(jdbcuserkerberos, "databases.presto.jdbc_user is null");
+        checkTableOwner(jdbcuserkerberos);
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    // Without hdfs impersonation, the table should be created with user specified for the
+    // principal hive.hdfs.presto.principal in hive.properties
+    @Test(groups = {SINGLENODE_KERBEROS_HDFS_NO_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeKerberosHdfsNoImpersonation()
+            throws Exception
+    {
+        checkTableOwner("hdfs");
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    private void checkTableOwner(String expectedOwner)
+    {
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+        query(format("create table %s as select * from nation", tableName));
+        String owner = hdfsClient.getOwner(tablepath);
+        assertEquals(owner, expectedOwner);
```

Comment:
I'd consider adding `drop table` to @AfterTestWithContext annotated method instead, to avoid leaving unclean environment after test run when some exception is thrown before `drop  table` is executed.